File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/comment/dto/CommentResDto.java
package io.github.hs96wings.streaming_server.comment.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import io.github.hs96wings.streaming_server.comment.domain.Comment;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CommentResDto {
    private Long id;
    private String authorName;
    private String content;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;

    public CommentResDto(Comment comment) {
        this.id = comment.getId();
        this.authorName = comment.getAuthor().getUserid();
        this.content = comment.getContent();
        this.createdAt = comment.getCreatedAt();
    }
}

File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/comment/dto/CommentResDto.kt
package io.github.hs96wings.streaming_server.comment.dto

import com.fasterxml.jackson.annotation.JsonFormat
import io.github.hs96wings.streaming_server.comment.domain.Comment
import io.github.hs96wings.streaming_server.common.domain.BaseEntity
import java.time.LocalDateTime

data class CommentResDto(
    val id: Long,
    val authorName: String,
    val content: String,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val createdAt: LocalDateTime?
) {
    constructor(comment: Comment): this (
        id = comment.id ?: throw IllegalStateException("Comment ID cannot be null for DTO conversion"),
        authorName = comment.author.userid,
        content = comment.content,
        createdAt = comment.createdAt
    )
}